### PR TITLE
Update upgrade cluster

### PR DESCRIPTION
**Pull request by ngaumont**
This adds two commands:
- `cb upgrade update`
- `cb maintenance update`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 - `cb list` command now accepts `--format`. Supports: `table` and `tree`.
 - `cb list` command now accepts `--team`.
+- `cb maintenance update` command added to update a pending maintenance.
+- `cb upgrade update` command added to update a pending upgrade.
```
Comment (abrightwell) [inline CHANGELOG.md:12]: Once #106 is merged, let's be sure to regroup these.

File: src/cli.cr [hunk 2]
```diff
@@ -204,6 +204,33 @@ op = OptionParser.new do |parser|
...
+        $ cb upgrade update --cluster <ID> --storage 800
+
+        Update a pending upgrade to a new plan
+        $ cb upgrade update --cluster <ID> --plan memory-128
+
+        Update a pending upgrade to failover as soon as possible
+        $ cb upgrade update --cluster <ID> --now
+
+        Update a upgrade to failover starting from a given time
+        $ cb upgrade update --cluster <ID> --starting-from 2022-01-01T00:00:00Z
```
Comment (abrightwell) [inline src/cli.cr:230]:
```suggestion
        Update a pending upgrade to a new storage size.
        $ cb upgrade update --cluster <ID> --storage 800

        Update a pending upgrade to a new plan.
        $ cb upgrade update --cluster <ID> --plan memory-128

        Update a pending upgrade to failover as soon as possible.
        $ cb upgrade update --cluster <ID> --now

        Update a upgrade to failover starting from a given time.
        $ cb upgrade update --cluster <ID> --starting-from 2022-01-01T00:00:00Z
```

File: src/cli.cr [hunk 3]
```diff
@@ -329,6 +356,27 @@ op = OptionParser.new do |parser|
...
+
+      parser.examples = <<-EXAMPLES
+        Update a pending maintenance to failover during the cluster maintenance window
+        $ cb maintenance update --cluster <ID> --use-cluster-maitenance-window
+
+        Update a pending maintenance to failover as soon as possible
+        $ cb maintenance update --cluster <ID> --now
+
+        Update a maintenance to failover starting from a given time
+        $ cb maintenance update --cluster <ID> --starting-from 2022-01-01T00:00:00Z
```
Comment (abrightwell) [inline src/cli.cr:377]:
```suggestion
        Update a pending maintenance to failover during the cluster maintenance window.
        $ cb maintenance update --cluster <ID> --use-cluster-maintenance-window

        Update a pending maintenance to failover as soon as possible.
        $ cb maintenance update --cluster <ID> --now

        Update a maintenance to failover starting from a given time.
        $ cb maintenance update --cluster <ID> --starting-from 2022-01-01T00:00:00Z
```

**Review by abrightwell — APPROVED** [review]
Just a few minor updates requested. Otherwise, LGTM.

File: CHANGELOG.md [hunk 4]
```diff
@@ -8,6 +8,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 - `cb list` command now accepts `--format`. Supports: `table` and `tree`.
 - `cb list` command now accepts `--team`.
+- `cb maintenance update` and `cb upgrade update` update a pending maintenance
+  and a pending upgrade respectively.
```
Comment (ngaumont) [inline CHANGELOG.md:12]: Sorry to bother you with changelog update @abrightwell (and all the maintenance typo…).
Is this fine for you ?

Comment (abrightwell) [reply CHANGELOG.md:12]: WFM

**Review by abrightwell — APPROVED** [review]

**Pull request merged**